adding counters for Prometeus

MihaelaHristov opened the pull request.

MihaelaHristov committed:
adding counters for Prometeus

simoonH committed:
Merge branch 'main' into w08-Prometeus-stuff

MihaelaHristov committed:
added timer for each db operation for Prometeus monitoring

MihaelaHristov committed:
addMessages monitoring only

MihaelaHristov committed:
without db_monitoring

MihaelaHristov committed:
docker compose pirt for grafana should always be 3000:3000

MihaelaHristov committed:
added monitoring for hte rest of db methods

MihaelaHristov committed:
now active users makes more sence

MihaelaHristov committed:
adding label by function for DB monitoring

MihaelaHristov committed:
forgot to register dbProcessDuration with Prometeus registry

MihaelaHristov committed:
testing for active users metric

dudustri commented on remote_files/docker-compose.yml:
```diff
@@ -42,7 +50,7 @@ services:
     image: grafana/grafana:10.2.4
     ports:
-      - "3001:3000"
+      - "3000:3000"
```
Is it correct? why should we change the VM port?

TomasBabkine-DiCaprio replied:
I'm curious as to why it was 3001:3000, does that mean "either 3001 or 3000"? Where was Grafana listening before? 

Either way, I don't think changing the port makes a big difference but maybe I am wrong

dudustri replied:
It will make no difference but I'm curious why it was 3001 👴

dudustri reviewed (approved):
Very good! Looks cool with all the monitor variables you added 👌😁

TomasBabkine-DiCaprio commented on remote_files/docker-compose.yml:
```diff
@@ -102,14 +114,15 @@ services:
       - elasticsearch # Ensures that Kibana starts after Elasticsearch.
     ports:
-      - "5601:5601"  # Exposes Kibana's web interface port.
+      - "5601:5601" # Exposes Kibana's web interface port.
```
Does this mean in the range from 5601 to 5601? Can this be simplified to just 5601?

MihaelaHristov replied:
3000:3000 (port on host machine : port inside Docker container) its not a range, but a mapping of host to container, Technically it should work if 3001 (if that port is not used from other application already), same with 5601:5601
For me it fails when it is 3001:3000

TomasBabkine-DiCaprio replied:
Ah makes sense! Thank you for the clarifiction :)

TomasBabkine-DiCaprio reviewed (changes requested):
I think we could improve this PR with a few things:
- Add units to the counters (seconds, milliseconds, etc)
- Add less ambiguous description for some of the counters
- Rework how the timers work...they seem oddly complicated. Or at least understand how they work, we shouldn't have code that we don't understand in production.

I can help with all of these this week :)

MihaelaHristov commented:
### Add units:
- I have cleaned the grafana dashboard by excluding sum  because it is just confusing 
- added in the title the quiry metric incorporating units and label, so it must be more clear

### Add less ambiguous description:
not sure if you mean the title ?- that one is updated in Grafana now

### Rework how the timers work...they seem oddly complicated
- please read the clarification in knowledge base under week06/ 1) Add Monitoring to your systems
- it incorporates links to official documentation from 'GOprometeus' and clear explanation (imo)
- I am not sure if it can be done in more simplified way that this but if you have other idea I am open to it :)
